exp/term/ansi/wrap.go

```diff
@@ -227,6 +229,153 @@ func Wordwrap(s string, limit int, breakpoints string) string {
 	return buf.String()
 }
 
+// SmartWrap wraps a string or a block of text to a given line length, breaking
+// word boundaries if necessary. This will preserve ANSI escape codes and will
+// account for wide-characters in the string. The breakpoints string is a list
+// of characters that are considered breakpoints for word wrapping. A hyphen
+// (-) is always considered a breakpoint.
+func SmartWrap(s string, limit int, breakpoints string) string {
```

Comment:
Maybe `SmartWrap` is incorrect since there's a smarter way to wrap the text with less word breakage. Perhaps `Wrap` should be `Hardwrap` and this becomes just `Wrap`.